lib/Default/Template.class.inc

```diff
@@ -31,7 +31,9 @@ class Template {
 	}
 	
 	public function display($templateName,$forAjax=false, $xmlResponse=false) {
-		@ob_clean();
+		if (ob_get_length() > 0) {
+			ob_clean();
```

Comment:
Since this appeared, I added this in order to get rid of any junk generated by files before reaching the templates, which is also why none of the files had any whitespace after the closing `?>` - I'd forgotten the reasoning for removing that (which can be tough as linux programs like to enforce an empty line, because a lot of linux programs used to break if it was omitted I believe) - but it would actually make sense to remove the closing tags altogether to avoid that problem and hopefully eventually make it possible to get rid of this call, which is really just a hack to deal with the older code in SMR

Reply:
I'm not sure I understand your recommendation here. Are you suggesting we remove the trailing `?>` from every file and then remove `ob_clean();` here?

Reply:
Yeah, the trailing `?>` should be removed - check out https://secure.php.net/manual/en/language.basic-syntax.phptags.php
> If a file is pure PHP code, it is preferable to omit the PHP closing tag at the end of the file. This prevents accidental whitespace or new lines being added after the PHP closing tag, which may cause unwanted effects because PHP will start output buffering when there is no intention from the programmer to send any output at that point in the script.

As for removing the `ob_clean` the reason we need it at all is because we use output buffering in order to stop
```
?>

```
from sending the trailing newline immediately which was breaking things like redirects (because the redirect is a header and must be sent before any output).  So for that I'd recommend turning it off in dev mode but leaving it on for prod so that it can be found/fixed without breaking prod (like how you enable warnings in dev but not in prod)